src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -617,17 +615,24 @@ class DateBucketPartitioner(gran: Granularity, intervals: Iterable[Interval]) ex
   }
 }
 
-class DateBucketAndHashPartitioner(gran: Granularity, partMap: Map[(Long, Long), Int], optionalDims : Option[Set[String]] = None)
+class DateBucketAndHashPartitioner(gran: Granularity, partMap: Map[(Long, Long), Int], optionalDims: Option[Set[String]] = None)
   extends Partitioner {
-  val maxTimePerBucket = partMap.groupBy(_._1._1).map(e => e._1 -> e._2.size)
+  lazy val shardLookups = partMap.groupBy(_._1._1)
+    .map(e => e._1 -> e._2.size)
+    .mapValues(x => {
+      val shardSpecTemplate = new HashBasedNumberedShardSpec(0, x, SerializedJsonStatic.mapper)
+      shardSpecTemplate.getLookup(Range(0, x)
+        .map(y => new HashBasedNumberedShardSpec(y, x, SerializedJsonStatic.mapper).asInstanceOf[ShardSpec])
+        .toList.asJava
+      )
+    })
   val dims = optionalDims.orNull
```

Comment:
we went through the trouble of using option types, so why do we use null instead here to indicate missing value ?